odoo/tools/translate.py

```diff
@@ -888,6 +891,72 @@ def handle_text(text, lineno):
     _extract_translatable_qweb_terms(tree.getroot(), handle_text)
     return result
 
+
+def extract_formula_terms(formula):
+    """Extract strings in a spreadsheet formula which are arguments to '_t' functions
+    >>> extract_formula_terms('=_t("Hello") + _t("Raoul")')
+    ["Hello", "Raoul"]
+    """
+    terms = []
+    tokens = tokenize(io.BytesIO(formula.encode('utf-8')).readline)
+    tokens = list(token for token in tokens if token.type not in {NEWLINE, INDENT, DEDENT} )
+    for i in range(len(tokens)):
+        next_four = tokens[i:i+4]
```

Comment:
The loop can probably pull straight from the tokens iterator:

- iterate on tokens
- check if current is `_t`
  - if not continue
- get and check that the next token is `(`
- get next token
- get and check that the next token is `)`
- yield the 3rd token

Not sure what should be done for a `_t` which is not a function call though, does that make any sense? Should the extractor just bail and return? Or raise?

Reply:
much better like that indeed :)
Thanks for the reviews!

> Not sure what should be done for a _t which is not a function call though

I'd say: ignore it here. It's invalid but will be gracefully handled by the o-spreadsheet tokenizer and parser